internal/config/watchd/watchd.go

```diff
@@ -70,31 +83,36 @@ func WriteConfig(confFile string, dirs []string) error {
 	cfg := AppConfig{Dirs: dirs, Verbose: 0}
 	data, err := yaml.Marshal(&cfg)
 	if err != nil {
-		return fmt.Errorf(i18n.G("unable to marshal config: %v"), err)
+		return fmt.Errorf(i18n.G("unable to marshal: %v"), err)
 	}
 
 	if err := os.WriteFile(confFile, data, 0600); err != nil {
-		return fmt.Errorf(i18n.G("unable to write config file: %v"), err)
+		return err
 	}
 
 	return nil
 }
 
-// GetConfigFileFromArgs returns the path to the config file extracted from the
+// ConfigFileFromArgs returns the path to the config file extracted from the
 // command line arguments.
 //
 // This is not an exhaustive implementation of "parsing" the command line and
 // only covers the cases used by the service installer, which should be good
 // enough for us.
-func GetConfigFileFromArgs(args string) (string, error) {
+func ConfigFileFromArgs(args string) (string, error) {
+	err := fmt.Errorf(i18n.G("missing config file in CLI arguments"))
+
 	_, configFile, found := strings.Cut(args, "-c")
 	if !found {
-		return "", fmt.Errorf(i18n.G("missing config file in CLI arguments"))
+		return configFile, err
 	}
 
 	// Remove trailing quotes and spaces (quotes are added if the path to the
 	// config file contains spaces)
 	configFile = strings.Trim(configFile, `" `)
+	if configFile == "" {
+		return configFile, err
```

Comment:
I find: `return "", err` more readable (classic usage of returning the default value, instead of wondering why we return configFile here to have to read the line above… Ah it’s empty). Less cognitive load :p